apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLogger.cs

```diff
@@ -0,0 +1,178 @@
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
+        public static ErrorReportingExceptionLogger Create(
+            ReportErrorsServiceClient client, string projectId, string serviceName, string version) =>
+            Create(Task.FromResult(client), projectId, serviceName, version);
+
+        /// <summary>
+        /// Creates an instance of <see cref="ErrorReportingExceptionLogger"/>
+        /// </summary>
+        /// <param name="clientTask">The Error Reporting client.</param>
+        /// <param name="projectId">The Google Cloud Platform project ID.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
+        public static ErrorReportingExceptionLogger Create(
+            Task<ReportErrorsServiceClient> clientTask, string projectId, string serviceName, string version) =>
+            new ErrorReportingExceptionLogger(clientTask, projectId, serviceName, version);
+
+        /// <summary>
+        /// Creates an instance of <see cref="ErrorReportingExceptionLogger"/> using credentials as
+        /// defined by <see cref="GoogleCredential.GetApplicationDefaultAsync"/>.
+        /// </summary>
+        /// <param name="projectId">The Google Cloud Platform project ID.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
+        public static ErrorReportingExceptionLogger Create(
+            string projectId, string serviceName, string version) =>
+            Create(ReportErrorsServiceClient.CreateAsync(), projectId, serviceName, version);
+
+        private ErrorReportingExceptionLogger(
+           Task<ReportErrorsServiceClient> clientTask, string projectId, string serviceName, string version)
+        {
+            GaxPreconditions.CheckNotNull(projectId, nameof(projectId));
+            _clientTask = GaxPreconditions.CheckNotNull(clientTask, nameof(clientTask));
+            _projectName = new ProjectName(projectId);
+            _serviceContext = new ServiceContext
+            {
+                Service = GaxPreconditions.CheckNotNull(serviceName, nameof(serviceName)),
+                Version = GaxPreconditions.CheckNotNull(version, nameof(version)),
+            };
+        }
+
+        /// <summary>
+        /// Asynchronously reports an exception that occurred to the Stackdriver Error Reporting API.
+        /// </summary>
+        public async Task<ReportErrorEventResponse> ReportAsync(HttpContext context, Exception exception)
```

Comment:
You may want to add `<returns>` tags to describe the return value (same comment below).